### PR TITLE
refactor(pkg): refactor logic of block_batch_iterator 

### PR DESCRIPTION
**Pull request by YoGhurt111**
### CHANGELOGS
1. add DefaultBlockConfirmations in block_batch_iterator
2. remove unused code

### CODE REVIEW

**Comment by davidtaikocha** [comment]
Looks like there is a lint error?

**Comment by YoGhurt111** [comment]
> Looks like there is a lint error?

Yes, I will check this

File: pkg/chain_iterator/block_batch_iterator.go [hunk 1]
```diff
@@ -174,9 +175,16 @@ func (i *BlockBatchIterator) iter() (err error) {
 	if i.endHeight != nil {
 		destHeight = *i.endHeight
 	} else {
-		if destHeight, err = i.client.BlockNumber(i.ctx); err != nil {
+		destHeight, err = i.client.BlockNumber(i.ctx)
+		if err != nil {
 			return err
 		}
+		if destHeight > DefaultBlockConfirmations {
```
Comment (davidtaikocha) [inline pkg/chain_iterator/block_batch_iterator.go:182]: Looks like the confirmations are not configurable at all, so its always `0`?

Comment (YoGhurt111) [reply pkg/chain_iterator/block_batch_iterator.go:182]: fix it later

File: cmd/flags/prover.go [hunk 2]
```diff
@@ -172,6 +172,13 @@ var (
 		Usage:    "Version or tag or the L2 Node Version used as an L2 RPC Url by this guardian prover",
 		Category: proverCategory,
 	}
+	// Confirmations specific flag
+	BlockConfirmations = &cli.Uint64Flag{
+		Name:     "prover.blockConfirmations",
+		Usage:    "Confirmations to the latest l1 block before submitting a proof for a l2 block",
+		Value:    6,
```
Comment (mask-pp) [inline cmd/flags/prover.go:179]:
```suggestion
		Value:    0,
```

File: pkg/chain_iterator/block_batch_iterator.go [hunk 3]
```diff
@@ -19,6 +19,7 @@ import (
 const (
 	DefaultBlocksReadPerEpoch = 1000
 	DefaultRetryInterval      = 12 * time.Second
+	DefaultBlockConfirmations = 0
```
Comment (mask-pp) [inline pkg/chain_iterator/block_batch_iterator.go:22]: If BlockConfirmations is not set default value is 0, so this field is not necessary.

File: pkg/chain_iterator/block_batch_iterator.go [hunk 4]
```diff
@@ -54,6 +55,7 @@ type BlockBatchIterator struct {
 	isEnd              bool
 	reorgRewindDepth   uint64
 	retryInterval      time.Duration
+	blockConfirmations *uint64
```
Comment (mask-pp) [inline pkg/chain_iterator/block_batch_iterator.go:58]:
```suggestion
	blockConfirmations uint64
```

File: pkg/chain_iterator/block_batch_iterator.go [hunk 5]
```diff
@@ -65,6 +67,7 @@ type BlockBatchIteratorConfig struct {
 	OnBlocks              OnBlocksFunc
 	ReorgRewindDepth      *uint64
 	RetryInterval         time.Duration
+	BlockConfirmations    *uint64
```
Comment (mask-pp) [inline pkg/chain_iterator/block_batch_iterator.go:70]: ditto

File: pkg/chain_iterator/block_batch_iterator.go (hunk 3 above)
Comment (YoGhurt111) [reply pkg/chain_iterator/block_batch_iterator.go:22]: ok

File: pkg/chain_iterator/block_batch_iterator.go (hunk 5 above)
Comment (YoGhurt111) [reply pkg/chain_iterator/block_batch_iterator.go:70]: It is not necessary to add this in BlockBatchIteratorConfig. There is no use in this struct, do I still need to add it to this struct?

**Review by mask-pp — APPROVED** [review]

**Pull request merged**